src/lib/pipelines/variableGroup.test.ts

```diff
@@ -406,97 +405,76 @@ describe("authorizeAccessToAllPipelines", () => {
 
 describe("buildVariablesMap", () => {
   test("should create variable map with two variables", async () => {
-    const var1: IVariablesMap = {
+    const variables: IVariableGroupDataVariable = {
       var1: {
         isSecret: false,
         value: "val1"
-      }
-    };
-    const var2: IVariablesMap = {
+      },
       var2: {
         isSecret: true,
         value: "val2"
       }
     };
 
-    const variables: IVariablesMap[] = [];
-    variables.push(var1);
-    variables.push(var2);
-
-    let map: IVariablesMap | undefined;
-    const expectedMap: any = `{"0":{"var1":{"isSecret":false,"value":"val1"}},"1":{"var2":{"isSecret":true,"value":"val2"}}}`;
-    map = await buildVariablesMap(variables);
+    const expectedMap =
+      '{"var1":{"isSecret":false,"value":"val1"},"var2":{"isSecret":true,"value":"val2"}}';
+    const map = await buildVariablesMap(variables);
     expect(JSON.stringify(map)).toEqual(expectedMap);
     logger.info(`map: ${JSON.stringify(map)}`);
   });
 
   test("should create variable map with one variable", async () => {
-    const var1: IVariablesMap = {
+    const variables: IVariableGroupDataVariable = {
       var1: {
         isSecret: false,
         value: "val1"
       }
     };
 
-    const variables: IVariablesMap[] = [];
-    variables.push(var1);
-
-    let map: IVariablesMap | undefined;
-    const expectedMap: any = `{"0":{"var1":{"isSecret":false,"value":"val1"}}}`;
-    map = await buildVariablesMap(variables);
+    const expectedMap: any = `{"var1":{"isSecret":false,"value":"val1"}}`;
+    const map = await buildVariablesMap(variables);
     expect(JSON.stringify(map)).toEqual(expectedMap);
```

Comment:
preferred not to use `any` and do object compare. Can we do this?
```
const variables: IVariableGroupDataVariable = {
  var1: {
    isSecret: false, 
    value: "val1"
  }
};
const map = await buildVariablesMap(variables);
expect(map)).toEqual(variables);
```

Reply:
Makes sense!